docs: update rancher integration rfc (#49)
kubewarden/rfc: rfc/0010-rancher-integration-policies.md

This RFC explains how the Kubewarden policies, either owned by the Kubewarden
team or 3rd party ones, can be integrated in Rancher Explorer.

> For policies owned by the Kubewarden team it proposes building a catalog,
> using the [Rancher's Helm charts catalog system](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/helm-charts-in-rancher).
> 
> For 3rd party policies not owned by the Kubewarden team,
> it proposes using the Artifact Hub API to poll for the metadata stored in

Reviewer: To be completely specific about what 'it' is.
```suggestion
For policies owned by the Kubewarden team this RFC proposes building a catalog,
using the [Rancher's Helm charts catalog system](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/helm-charts-in-rancher).

For 3rd party policies not owned by the Kubewarden team,
this RFC proposes using the Artifact Hub API to poll for the metadata stored in
```

   in a simple way. For example: "PSP best of", "Security best practices",...

> 3. As a user, I want to install 3rd party Kubewarden policies via Rancher

Reviewer: These next three read very slightly better with 'the'.
```suggestion
3. As a user, I want to install 3rd party Kubewarden policies via the Rancher
```

   Explorer UI.
4. As a Kubewarden developer, I want to release a new Kubewarden policy so

>    it can be installed via Rancher Explorer UI.

Reviewer: ```suggestion
   it can be installed via the Rancher Explorer UI.
```

5. As a 3rd party developer, I want to release a policy on Artifact Hub so it can

>    be discovered and installed via Rancher Explorer UI.

Reviewer: ```suggestion
   be discovered and installed via the Rancher Explorer UI.
```

can edit the GitHub draft Release title by hand, close the PR, delete its
branch, and dispatch the PR job manually again.

### Policy catalog

Once released, the policies are consumed by following the process listed in
[RFC-9, Rancher integration of Kubewarden
charts](./0009-rancher-integration-charts.md) with the following changes.

All policies will be listed in a new repository:  
[https://github.com/kubewarden/policy-catalog](https://github.com/kubewarden/policy-catalog).  
This repository will act as a Helm chart repository, published via GitHub Pages.

Each policy release will generate a `Chart.yaml` file that includes the metadata required by the Rancher UI.
This file will be placed at `charts/<policy-name>/Chart.yaml`.
If the original policy contains a `questions-ui.yml` file, it will be copied to `charts/policy-name/questions.yaml`.
The `LICENSE` and `README.md` files will also be copied into the chart directory.

This is not a functional Helm chart, as it doesn't include any templates or values.

> It simply provides the metadata needed by the Rancher Explorer UI, making use of the existing Helm chart repository structure.

Reviewer: ```suggestion
It only provides the metadata needed by the Rancher Explorer UI, making use of the existing Helm chart repository structure.
```

### Chart.yaml generation

For each policy release the `Chart.yaml` file will be generated or updated from the `artifacthub-pkg.yml` and `artifacthub-repo.yml` files present in the policy repository.

The `name`, `version`, `description`, `keywords` fields will be taken from the `artifacthub-pkg.yml` file.

All the annotations present in `artifacthub-pkg.yml` will be copied to the `Chart.yaml` file.
Addtionally, the following annotations will be added to the `Chart.yaml` file:

- `artifacthub.io/repository`: The name of the Artifact Hub repository where the policy is published. This will be fetched by querying the Artifact Hub API using the `repositoryID` present in the `artifacthub-repo.yml` file.
- `kubewarden/registry`: The OCI registry where the policy is published. This is obtained by the `containesImages` field in the `artifacthub-pkg.yml` file.
- `kubewarden/repository`: The OCI repository where the policy is published. This is obtained by the `containesImages` field in the `artifacthub-pkg.yml` file.
- `kubewarden/tag`: The OCI tag where the policy is published. This is obtained by the `containesImages` field in the `artifacthub-pkg.yml` file.

> - `kubewarde/displayName`: The display name of the policy. This is obtained by the `displayName` field in the `artifacthub-pkg.yml` file.

Reviewer 2: ```suggestion
- `kubewarden/displayName`: The display name of the policy. This is obtained by the `displayName` field in the `artifacthub-pkg.yml` file.
```
